### PR TITLE
Add context manager for selecting min mem cuda device

### PR DESCRIPTION
**Pull request by Balandat**
This is helpful for avoiding OOMs when running a bunch of cuda tests in parallel on a multi-GPU machine.

This is just a proof of concept, if this seems reasonable I'll add it to all of the cuda tests.

### CODE REVIEW

File: test/_utils.py [hunk 1]
```diff
@@ -17,3 +20,18 @@ def approx_equal(self, other, epsilon=1e-4):
             "Size mismatch between self ({self}) and other ({other})".format(self=self.size(), other=other.size())
         )
     return torch.max((self - other).abs()) <= epsilon
+
+
+def get_cuda_max_memory_allocations() -> int:
```
Comment (jacobrgardner) [inline test/_utils.py]: Is the typing Python 3.6 compatible? We don't yet require 3.7.

Comment (jacobrgardner) [reply test/_utils.py]: Other than that, this looks good to me!

Comment (Balandat) [reply test/_utils.py]: yeah typing is py36+

Comment (gpleiss) [reply test/_utils.py]: I think we have to be py3.5 compatible for the docs, but Travis isn't complaining...
I'll merge, but let's check that we haven't destroyed the docs in the process :)

**Comment by Balandat** [comment]
Shooting from the hip there are we? This was meant to be a draft PR for visibility. Still need to update all the other cuda tests. But hey let’s test in prod :)

**Comment by gpleiss** [comment]
Oooops my bad! At least it only affects the unit tests 😬

**Pull request merged**